precompiles/referenda/src/lib.rs

```diff
@@ -356,18 +374,41 @@ where
 	/// * index: The index of a closed referendum whose Decision Deposit has not yet been refunded.
 	#[precompile::public("refundDecisionDeposit(uint32)")]
 	fn refund_decision_deposit(handle: &mut impl PrecompileHandle, index: u32) -> EvmResult {
-		let event = log1(
-			handle.context().address,
-			SELECTOR_LOG_DECISION_DEPOSIT_REFUNDED,
-			EvmDataWriter::new().write::<u32>(index).build(),
-		);
-		handle.record_log_costs(&[&event])?;
+		handle.record_log_costs_manual(1, 32 * 3)?;
+		// Get refunding deposit before dispatch
+		handle.record_cost(RuntimeHelper::<Runtime>::db_read_gas_cost())?;
+		let refunded_deposit: U256 = match ReferendumInfoFor::<Runtime>::get(index)
+			.ok_or(RevertReason::custom("Referendum index does not exist").in_field("index"))?
+		{
+			ReferendumInfo::Ongoing(x) => {
+				if let Some(d) = x.decision_deposit {
+					d.amount.into()
+				} else {
+					U256::zero()
+				}
+			}
+			ReferendumInfo::Approved(_, _, Some(d))
+			| ReferendumInfo::Rejected(_, _, Some(d))
+			| ReferendumInfo::TimedOut(_, _, Some(d))
+			| ReferendumInfo::Cancelled(_, _, Some(d)) => d.amount.into(),
+			// We let the pallet handle the RenferendumInfo validation logic on dispatch.
+			_ => U256::zero(),
+		};
 
 		let origin = Runtime::AddressMapping::into_account_id(handle.context().caller);
 
 		let call = ReferendaCall::<Runtime>::refund_decision_deposit { index }.into();
 
 		<RuntimeHelper<Runtime>>::try_dispatch(handle, Some(origin).into(), call)?;
+		let event = log1(
+			handle.context().address,
+			SELECTOR_LOG_DECISION_DEPOSIT_REFUNDED,
+			EvmDataWriter::new()
+				.write::<u32>(index)
+				.write::<Address>(Address(handle.context().caller))
```

Comment:
To match the substrate events, this should NOT be the caller, it should be the account that received the refunded deposit, so you need to also get `d.who` in the match further above

Reply:
Thank you 🙏